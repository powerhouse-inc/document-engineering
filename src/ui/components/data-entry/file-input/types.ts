import type { InputHTMLAttributes } from 'react'
import type { InputBaseProps } from '../../../../scalars/components/types.js'

export interface FileInputProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'value' | 'defaultValue' | 'onChange'>,
    InputBaseProps<File> {
  allowedFileTypes?: string[]
  maxFileSize?: number
  dragAndDropEnabled?: boolean
  onChange?: (file: File | null) => void
  // Item Props
  fileName?: string
  mimeType?: string
  fileSize?: number
  progress?: number
  onCancel?: (e: React.MouseEvent<HTMLButtonElement>) => void
  onReload?: (e: React.MouseEvent<HTMLButtonElement>) => void
  errorsUpload?: string[]
  status?: UploadFile
  onPreview?: () => void
}

export type UploadFile = 'idle' | 'uploading' | 'success' | 'error'
