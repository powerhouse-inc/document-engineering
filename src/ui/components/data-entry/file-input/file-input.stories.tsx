import type { Meta, StoryObj } from '@storybook/react'
import {
  getDefaultArgTypes,
  getValidationArgTypes,
  PrebuiltArgTypes,
  StorybookControlCategory,
} from '../../../../scalars/lib/storybook-arg-types.js'
import { FileInput } from './file-input.js'

/**
 * ## FileInput Component
 *
 * `FileInput` is a UI component for uploading a single file, supporting drag-and-drop and manual selection.
 *
 * This component is designed for user-friendly file input in forms or upload workflows. It includes:
 *
 * - Drag-and-drop area
 * - File type and size restrictions
 * - Accessibility support (`aria-*`, keyboard navigation)
 * - Custom label, description, and styling support
 *
 * ### Features
 * - Accessible: follows ARIA guidelines
 * - Configurable: accepts file type filters and max file size
 * - Integrates with external form libraries via refs
 *
 * > **Note:** This is a _presentational_ component only. It does not include form validation or submission logic.
 * >
 * > If you need validation, use the [`FileField`](?path=/docs/scalars-file-field--readme) component, which wraps `FileInput` with validation support and form bindings.
 */
const meta: Meta<typeof FileInput> = {
  title: 'Data Entry/File Input',
  component: FileInput,
  parameters: {
    layout: 'centered',
    form: {
      resetBehavior: 'unmount',
    },
  },
  tags: ['autodocs'],
  argTypes: {
    ...getDefaultArgTypes(),
    ...PrebuiltArgTypes.className,
    value: {
      control: 'file',
      description: 'Current value of the input field',
      table: {
        type: { summary: 'File | null' },
        category: StorybookControlCategory.DEFAULT,
      },
    },
    defaultValue: {
      control: 'file',
      description: 'Default value for the input field',
      table: {
        type: { summary: 'File | null' },
        category: StorybookControlCategory.DEFAULT,
      },
    },
    allowedFileTypes: {
      control: 'object',
      description: 'Allowed file types',
      table: {
        type: { summary: 'string[]' },
        category: StorybookControlCategory.DEFAULT,
      },
    },
    maxFileSize: {
      control: 'number',
      description: 'Maximum file size',
      table: {
        type: { summary: 'number' },
        category: StorybookControlCategory.DEFAULT,
      },
    },
    dragAndDropEnabled: {
      control: 'boolean',
      defaultValue: true,
      description: 'Drag and drop enabled',
      table: {
        defaultValue: { summary: 'true' },
        type: { summary: 'boolean' },
        category: StorybookControlCategory.DEFAULT,
      },
    },
    ...getValidationArgTypes({
      enabledArgTypes: {
        validators: false,
        showErrorOnBlur: false,
        showErrorOnChange: false,
      },
    }),
  },
  args: {
    name: 'file-input',
    dragAndDropEnabled: true,
  },
}

export default meta
type Story = StoryObj<typeof FileInput>

export const Default: Story = {
  args: {
    name: 'file-input',
    label: 'Upload File',
    description: 'Click to chose files',
    allowedFileTypes: ['xls', 'xlsx', 'png', 'docs'],
    maxFileSize: 15728640,
  },
}
