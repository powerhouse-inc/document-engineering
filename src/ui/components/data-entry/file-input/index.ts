export { FileInput } from './file-input.js'
export { type FileInputProps, type UploadFile } from './types.js'
