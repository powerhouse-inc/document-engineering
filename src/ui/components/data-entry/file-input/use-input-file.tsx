import { type DragEvent, useCallback, useState } from 'react'

export interface FileUploadOptions {
  value?: File
  defaultValue?: File
}

export const useInputFile = ({ value, defaultValue }: FileUploadOptions) => {
  const [file, setFile] = useState<File | null>(value ?? defaultValue ?? null)

  const handleDragEnter = useCallback((e: DragEvent<HTMLElement>) => {
    e.preventDefault()
    e.stopPropagation()
    setFile(e.dataTransfer.files[0])
  }, [])

  const handleDragLeave = useCallback((e: DragEvent<HTMLElement>) => {
    e.preventDefault()
    e.stopPropagation()

    if (e.currentTarget.contains(e.relatedTarget as Node)) {
      return
    }
    setFile(null)
  }, [])

  const handleDragOver = useCallback((e: DragEvent<HTMLElement>) => {
    e.preventDefault()
    e.stopPropagation()
  }, [])

  const handleDrop = useCallback((e: DragEvent<HTMLElement>) => {
    e.preventDefault()
    e.stopPropagation()
    setFile(e.dataTransfer.files[0])

    const files = e.dataTransfer.files
    if (files.length > 0) {
      // Only use the first file
      const file = files[0]
      setFile(file)
    }
  }, [])

  const openFileDialog = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation()
  }, [])

  return {
    file,
    handleDragEnter,
    handleDragLeave,
    handleDragOver,
    handleDrop,
    openFileDialog,
  }
}
