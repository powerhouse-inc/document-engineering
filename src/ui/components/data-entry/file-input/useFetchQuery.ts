import { useEffect, useState } from 'react'

interface UseQueryOptions {
  enabled?: boolean
}

interface UseQueryResult<T> {
  data: T | null
  isLoading: boolean
  error: unknown
}

const useFetchQuery = <T>(queryFn: () => Promise<T>, { enabled = true }: UseQueryOptions = {}): UseQueryResult<T> => {
  const [data, setData] = useState<T | null>(null)
  const [isLoading, setIsLoading] = useState(false)
  const [error, setError] = useState<unknown>(null)

  useEffect(() => {
    if (!enabled) {
      return
    }

    const fetchData = async () => {
      setIsLoading(true)
      setError(null)
      setData(null)

      try {
        const result = await queryFn()
        setData(result)
      } catch (err) {
        setError(err)
      } finally {
        setIsLoading(false)
      }
    }

    void fetchData()
  }, [queryFn, enabled])

  return { data, isLoading, error }
}

export default useFetchQuery
