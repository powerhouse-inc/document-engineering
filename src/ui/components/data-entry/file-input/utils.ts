export const formatBytes = (bytes?: number, decimals = 2, binaryUnits = false): string => {
  if (!bytes) return '0 Bytes'
  if (bytes === 0) return '0 Bytes'
  if (isNaN(bytes) || bytes < 0) return 'Invalid size'

  const base = binaryUnits ? 1024 : 1000
  const units = binaryUnits ? ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB'] : ['Bytes', 'kB', 'MB', 'GB', 'TB', 'PB']

  const dm = Math.max(0, decimals)

  const i = Math.min(Math.floor(Math.log(bytes) / Math.log(base)), units.length - 1)

  const num = bytes / base ** i
  const formatted = Number(num.toFixed(dm))

  return `${formatted} ${units[i]}`
}
