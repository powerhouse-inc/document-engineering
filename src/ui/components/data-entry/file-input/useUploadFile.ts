import { useCallback, useEffect, useState } from 'react'
import { detectPreviewType, getPreviewComponentFromError, validateFileForPreview } from './utils.js'
import useFetchQuery from './useFetchQuery.js'
import type { PreviewStatus } from './types.js'

interface UseFileUploadProps {
  value?: File | null
  defaultValue?: File | null
  onChange?: (file: File | null) => void
}

export const useFileUpload = ({ value, defaultValue, onChange }: UseFileUploadProps) => {
  const [file, setFile] = useState<File | null>(value ?? defaultValue ?? null)
  const [preview, setPreview] = useState<string | undefined>(undefined)
  const [isPreviewOpen, setIsPreviewOpen] = useState(false)

  const queryFn = useCallback(async () => {
    if (!file) return Promise.reject(new Error('No hay archivo seleccionado'))
    return validateFileForPreview(file)
  }, [file])
  const { data, isLoading, error } = useFetchQuery(queryFn, {
    enabled: isPreviewOpen && !!file,
  })

  const getPreviewStatus = (): PreviewStatus => {
    if (!isPreviewOpen) return 'idle'
    if (isLoading) return 'loading'
    if (error) {
      const errorType = getPreviewComponentFromError(error)
      return errorType || 'unsupported_file'
    }
    if (data === 'success') return 'success'
    return 'idle'
  }
  const previewStatus = getPreviewStatus()

  const handleDrop = useCallback(
    (acceptedFiles?: File[]) => {
      if (!acceptedFiles || acceptedFiles.length === 0) return
      const file = acceptedFiles[0]
      const preview = URL.createObjectURL(file)
      setPreview(preview)
      setFile(file)
      onChange?.(file)
    },
    [onChange]
  )

  const handleCancelPreview = () => {
    setIsPreviewOpen(false)
  }

  const typePreview = detectPreviewType(file ?? undefined)

  useEffect(() => {
    return () => {
      if (preview) {
        URL.revokeObjectURL(preview)
      }
    }
  }, [preview])

  const borderIndicator = file ? 'text-blue-900' : 'text-gray-300'

  const handleOnPreview = () => {
    if (!file) return
    setIsPreviewOpen(true)
  }

  return {
    handleDrop,
    file,
    borderIndicator,
    // Preview
    preview,
    handleCancelPreview,
    handleOnPreview,
    isPreviewOpen,
    previewStatus,
    typePreview,
  }
}
