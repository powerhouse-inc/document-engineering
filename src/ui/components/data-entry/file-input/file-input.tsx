import React, { useCallback, useId } from 'react'
import type { FileInputProps } from './types.js'
import { cn } from '../../../../scalars/lib/utils.js'
import { Icon } from '../../icon/icon.js'
import FileBackground from '../../icon-components/FileBackground.js'
import { Input } from '../../../../ui/components/data-entry/input/index.js'
import { FormLabel } from '../../../../scalars/components/fragments/form-label/form-label.js'
import { FormMessageList } from '../../../../scalars/components/fragments/form-message/index.js'
import { useDropzone } from 'react-dropzone'
import { formatBytes, getExtensionsFromMimeTypes } from './utils.js'
import { Button } from '../../../../scalars/components/fragments/button/button.js'
import { useFileUpload } from './useUploadFile.js'
import { FileUploadItem } from './sub-components/file-upload-item.js'

const FileInput = React.forwardRef<HTMLInputElement, FileInputProps>(
  (
    {
      name,
      label,
      description,
      required,
      disabled = false,
      errors = [],
      warnings = [],
      id: propId,
      maxFileSize,
      allowedFileTypes,
      value,
      defaultValue,
      dragAndDropEnabled = true,
      onChange,

      // Item Props
      fileName,
      fileSize,
      mimeType,
      progress,
      onCancel,
      onReload,
      errorsUpload,
      status = 'idle',
      ...props
    },
    ref
  ) => {
    const prefix = useId()
    const id = propId ?? `${prefix}-file`
    const hasError = Array.isArray(errors) && errors.length > 0
    const hasWarning = Array.isArray(warnings) && warnings.length > 0
    const allowedFileTypesString = Array.isArray(allowedFileTypes) ? allowedFileTypes : []

    const { handleDrop, borderIndicator } = useFileUpload({ value, defaultValue, onChange })

    const { getInputProps, getRootProps, open, inputRef } = useDropzone({
      onDropAccepted: (acceptedFiles) => {
        handleDrop(acceptedFiles)
      },
    })

    const mergedRef = useCallback(
      (node: HTMLInputElement | null) => {
        if (typeof ref === 'function') {
          ref(node)
        } else if (ref) {
          ref.current = node
        }

        ;(inputRef as React.MutableRefObject<HTMLInputElement | null>).current = node
      },
      [ref, inputRef]
    )

    return (
      <div className="flex flex-col">
        {!!label && (
          <FormLabel
            htmlFor={id}
            id={`${id}-label`}
            disabled={disabled}
            hasError={hasError}
            required={required}
            className={cn(disabled && 'pointer-events-none')}
          >
            {label}
          </FormLabel>
        )}

        <div className="flex flex-col w-full h-full pt-3.5 min-h-[148px]">
          <div className="relative h-[148px]">
            <div
              className={cn('absolute z-0 h-full w-full', borderIndicator)}
              data-state={value ? 'selected' : 'idle'}
              data-testid="file-input-border"
            >
              <FileBackground className="w-full h-full opacity-50" />
            </div>

            <div
              {...getRootProps({
                role: 'group',
                className: cn(
                  'dropzone',
                  'relative flex w-full flex-col h-full cursor-pointer',
                  !dragAndDropEnabled && 'opacity-50 pointer-events-none cursor-not-allowed',
                  disabled && 'cursor-not-allowed',
                  // padding
                  'px-3 py-4'
                ),
                tabIndex: disabled ? -1 : 0,
              })}
              data-testid="file-drop-area"
            >
              {status === 'idle' && (
                <div className="flex flex-col items-center justify-center gap-2">
                  <Icon name="FileUpload" size={40} className="text-gray-500" aria-hidden="true" />
                  <p className="text-gray-500 font-normal text-sm leading-5">{description}</p>
                </div>
              )}
              {status !== 'idle' && (
                <div className="absolute inset-0 flex items-center justify-center p-4">
                  <div className="w-full max-w-full">
                    <FileUploadItem
                      errorsUpload={errorsUpload}
                      mimeType={mimeType}
                      fileName={fileName}
                      fileSize={formatBytes(fileSize ?? 0, 2)}
                      progress={progress}
                      status={status}
                      onCancel={onCancel}
                      onReload={onReload}
                    />
                  </div>
                </div>
              )}
              <Input
                {...getInputProps({
                  name,
                  id,
                  required,
                  disabled,
                  type: 'file',
                  multiple: false,
                  ...props,
                })}
                aria-invalid={hasError}
                aria-required={required}
                aria-labelledby={label ? `${id}-label` : undefined}
                ref={mergedRef}
              />
            </div>
            {status === 'idle' && (
              <div
                className={cn(
                  'absolute bottom-4 left-1/2 transform -translate-x-1/2 z-1 pointer-events-auto',
                  disabled && 'pointer-events-none'
                )}
              >
                <Button
                  type="button"
                  onClick={open}
                  disabled={disabled}
                  aria-label="Search File"
                  className={cn(
                    'inline-block h-10 bg-white border border-[#E4E4E7] rounded-md text-gray-500 cursor-pointer text-sm font-medium leading-5',
                    // padding
                    'px-4 py-2',
                    // hover
                    'hover:bg-white hover:text-gray-500 hover:cursor-pointer'
                  )}
                >
                  Search File
                </Button>
              </div>
            )}
          </div>
        </div>
        <div className="flex flex-col gap-2 mt-2">
          <span className="text-gray-500 font-normal text-xs leading-4.5 truncate">
            Supports: {getExtensionsFromMimeTypes(allowedFileTypesString).join(', ')}
          </span>
          <span className="text-gray-500 font-medium text-xs leading-4.5 truncate">
            Max: {formatBytes(maxFileSize, 2)}
          </span>
        </div>
        <div className="flex flex-col gap-2 mt-2">
          {hasError && <FormMessageList messages={errors} type="error" />}
          {hasWarning && <FormMessageList messages={warnings} type="warning" />}
        </div>
      </div>
    )
  }
)

FileInput.displayName = 'FileInput'

export { FileInput }
