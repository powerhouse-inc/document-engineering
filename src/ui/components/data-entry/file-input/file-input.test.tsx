import { render, screen, waitFor } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import { FileInput } from './file-input.js'

describe('FileInput Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<FileInput name="file" label="Upload File" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render with label', () => {
    render(<FileInput name="file" label="Upload File" />)
    expect(screen.getByText('Upload File')).toBeInTheDocument()
  })

  it('should render with description', () => {
    render(<FileInput name="file" label="Upload File" description="Drop your file here or click to chose files" />)
    expect(screen.getByText('Drop your file here or click to chose files')).toBeInTheDocument()
  })

  it('should handle disabled state', () => {
    render(<FileInput name="file" label="Upload File" disabled />)
    expect(screen.getByLabelText('Upload File')).toBeDisabled()
  })

  it('should display erros messages', () => {
    render(<FileInput name="file" label="Upload File" errors={['Invalid file']} />)
    expect(screen.getByText('Invalid file')).toBeInTheDocument()
  })

  it('should have correct ARIA attributes', async () => {
    render(<FileInput name="file" label="Upload File" required errors={['Invalid file']} />)

    const input = screen.getByLabelText(/Upload File/)
    expect(input).toHaveAttribute('aria-required', 'true')
    await waitFor(() => {
      expect(input).toHaveAttribute('aria-invalid', 'true')
    })
  })

  it('should accept allowedFileTypes prop', () => {
    const allowedTypes = ['application/pdf', 'image/jpeg', 'image/png']
    render(<FileInput name="file" label="Upload File" allowedFileTypes={allowedTypes} />)

    expect(screen.getByText('Supports: pdf, jpg, png')).toBeInTheDocument()
  })

  it('should accept maxFileSize prop', () => {
    render(<FileInput name="file" label="Upload File" maxFileSize={10485760} />)

    expect(screen.getByText('Max: 10.49 MB')).toBeInTheDocument()
  })

  it('should disable drag and drop when dragAndDropEnabled is false', () => {
    render(<FileInput name="file" label="Upload File" dragAndDropEnabled={false} />)

    const dropArea = screen.getByTestId('file-drop-area')
    expect(dropArea).toHaveClass('pointer-events-none')
    expect(dropArea).toHaveClass('cursor-not-allowed')
  })
})
