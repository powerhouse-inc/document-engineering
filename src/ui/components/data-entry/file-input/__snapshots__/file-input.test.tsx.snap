// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FileInput Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <label
      class="inline-flex items-center text-sm font-medium leading-4 text-gray-900 dark:text-gray-50 mb-[3px]"
      for=":r0:-file"
      id=":r0:-file-label"
    >
      Upload File
    </label>
    <div
      class="flex flex-col w-full h-full pt-3.5 min-h-[148px]"
    >
      <div
        class="relative h-[148px]"
      >
        <div
          class="absolute z-0 h-full w-full text-gray-300"
          data-state="idle"
          data-testid="file-input-border"
        >
          <svg
            class="w-full h-full opacity-50"
            fill="none"
            viewBox="0 0 251 152"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.875 1V2H25.625V1H37.375V2H49.125V1H60.875V2H72.625V1H84.375V2H96.125V1H107.875V2H119.625V1H131.375V2H143.125V1H154.875V2H166.625V1H178.375V2H190.125V1H201.875V2H213.625V1H225.375V2H237.125V1H243C246.866 1 250 4.13401 250 8V13.667H249V25H250V36.333H249V47.667H250V59H249V70.333H250V81.667H249V93H250V104.333H249V115.667H250V127H249V138.333H250V144C250 147.866 246.866 151 243 151H237.125V150H225.375V151H213.625V150H201.875V151H190.125V150H178.375V151H166.625V150H154.875V151H143.125V150H131.375V151H119.625V150H107.875V151H96.125V150H84.375V151H72.625V150H60.875V151H49.125V150H37.375V151H25.625V150H13.875V151H8C4.13401 151 1 147.866 1 144V138.333H2V127H1V115.667H2V104.333H1V93H2V81.667H1V70.333H2V59H1V47.667H2V36.333H1V25H2V13.667H1V8L1.00879 7.63965C1.19633 3.94111 4.25486 1 8 1H13.875Z"
              fill="#F4F4F4"
            />
            <path
              d="M13.875 1V2H25.625V1H37.375V2H49.125V1H60.875V2H72.625V1H84.375V2H96.125V1H107.875V2H119.625V1H131.375V2H143.125V1H154.875V2H166.625V1H178.375V2H190.125V1H201.875V2H213.625V1H225.375V2H237.125V1H243C246.866 1 250 4.13401 250 8V13.667H249V25H250V36.333H249V47.667H250V59H249V70.333H250V81.667H249V93H250V104.333H249V115.667H250V127H249V138.333H250V144C250 147.866 246.866 151 243 151H237.125V150H225.375V151H213.625V150H201.875V151H190.125V150H178.375V151H166.625V150H154.875V151H143.125V150H131.375V151H119.625V150H107.875V151H96.125V150H84.375V151H72.625V150H60.875V151H49.125V150H37.375V151H25.625V150H13.875V151H8C4.13401 151 1 147.866 1 144V138.333H2V127H1V115.667H2V104.333H1V93H2V81.667H1V70.333H2V59H1V47.667H2V36.333H1V25H2V13.667H1V8L1.00879 7.63965C1.19633 3.94111 4.25486 1 8 1H13.875Z"
              stroke="currentColor"
              stroke-dasharray="12 12"
              stroke-width="2"
            />
          </svg>
        </div>
        <div
          class="dropzone relative flex w-full flex-col h-full cursor-pointer px-3 py-4"
          data-testid="file-drop-area"
          role="group"
          tabindex="0"
        >
          <div
            class="flex flex-col items-center justify-center gap-2"
          >
            <svg
              aria-hidden="true"
              class="text-gray-500"
              data-testid="icon-fallback"
              fill="none"
              height="40"
              style="width: 40px; height: 40px;"
              viewBox="0 0 41 40"
              width="41"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.5 4C26.134 4 29.33 7.846 29.794 12.492H29.934C33.564 12.492 36.5 15.516 36.5 19.246C36.5 19.4367 36.4927 19.6253 36.478 19.812C35.0422 18.0186 33.1023 16.6963 30.9081 16.0156C28.7139 15.3349 26.3662 15.3269 24.1675 15.9927C21.9687 16.6586 20.0198 17.9677 18.572 19.7513C17.1241 21.535 16.2436 23.7114 16.044 26H11.064C7.438 26 4.5 22.976 4.5 19.246C4.5 15.516 7.44 12.492 11.064 12.492H11.206C11.674 7.816 14.866 4 20.5 4Z"
                fill="#9EA0A1"
              />
              <path
                clip-rule="evenodd"
                d="M27.5 36C25.1131 36 22.8239 35.0518 21.136 33.364C19.4482 31.6761 18.5 29.3869 18.5 27C18.5 24.6131 19.4482 22.3239 21.136 20.636C22.8239 18.9482 25.1131 18 27.5 18C29.8869 18 32.1761 18.9482 33.864 20.636C35.5518 22.3239 36.5 24.6131 36.5 27C36.5 29.3869 35.5518 31.6761 33.864 33.364C32.1761 35.0518 29.8869 36 27.5 36ZM26.5003 25.718V31.126C26.5003 31.3912 26.6056 31.6455 26.7932 31.8331C26.9807 32.0206 27.235 32.126 27.5003 32.126C27.7655 32.126 28.0198 32.0206 28.2074 31.8331C28.3949 31.6455 28.5003 31.3912 28.5003 31.126V25.718L30.7043 27.922C30.9154 28.1331 31.2017 28.2517 31.5003 28.2517C31.7988 28.2517 32.0852 28.1331 32.2963 27.922C32.5074 27.7109 32.626 27.4245 32.626 27.126C32.626 26.8274 32.5074 26.5411 32.2963 26.33L28.2963 22.33C27.8563 21.89 27.1443 21.89 26.7043 22.33L22.7043 26.33C22.4932 26.5411 22.3745 26.8274 22.3745 27.126C22.3745 27.4245 22.4932 27.7109 22.7043 27.922C22.9154 28.1331 23.2017 28.2517 23.5003 28.2517C23.7988 28.2517 24.0852 28.1331 24.2963 27.922L26.5003 25.718Z"
                fill="#9EA0A1"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="text-gray-500 font-normal text-sm leading-5"
            />
          </div>
          <input
            hidden=""
            name="file"
            type="text"
          />
          <input
            aria-invalid="false"
            aria-labelledby=":r0:-file-label"
            class="flex h-9 w-full rounded-md text-sm font-normal leading-5 text-gray-900 dark:text-gray-50 dark:border-charcoal-700 dark:bg-charcoal-900 border border-gray-300 bg-white px-3 py-2 font-sans placeholder:text-gray-500 dark:placeholder:text-gray-600 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-gray-900 focus-visible:ring-offset-0 focus-visible:ring-offset-white dark:focus-visible:ring-charcoal-300 dark:focus-visible:ring-offset-charcoal-900 dark:focus:bg-charcoal-900 focus:bg-gray-50 disabled:cursor-not-allowed disabled:border-gray-300 disabled:bg-gray-50 disabled:text-gray-700 disabled:dark:border-charcoal-800 disabled:dark:bg-charcoal-900 disabled:dark:text-gray-300"
            id=":r0:-file"
            style="border: 0px; clip: rect(0, 0, 0, 0); clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            tabindex="-1"
            type="file"
          />
        </div>
        <div
          class="absolute bottom-4 left-1/2 transform -translate-x-1/2 z-1 pointer-events-auto"
        >
          <button
            aria-label="Search File"
            class="items-center justify-center gap-2 whitespace-nowrap transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 inline-block h-10 bg-white border border-[#E4E4E7] rounded-md text-gray-500 cursor-pointer text-sm font-medium leading-5 px-4 py-2 hover:bg-white hover:text-gray-500 hover:cursor-pointer"
            type="button"
          >
            Search File
          </button>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-2 mt-2"
    >
      <span
        class="text-gray-500 font-normal text-xs leading-4.5 truncate"
      >
        Supports: 
      </span>
      <span
        class="text-gray-500 font-medium text-xs leading-4.5 truncate"
      >
        Max: 0 Bytes
      </span>
    </div>
    <div
      class="flex flex-col gap-2 mt-2"
    />
  </div>
</DocumentFragment>
`;
